File: hello_mpi.c

#include <stdio.h>
#include <mpi.h>

int main (int argc, char * argv[])

{

    int rank, size;

    MPI_Init( &argc, &argv );

    MPI_Comm_rank( MPI_COMM_WORLD,&rank );

    MPI_Comm_size( MPI_COMM_WORLD,&size );

#ifdef __MIC
    printf( "MIC: Hello from processor %d of %d\n", rank, size );
#else
    printf( "CPU: Hello from processor %d of %d\n", rank, size );
#endif
    MPI_Finalize();

    return 0;
}

File: hello_mpi.f90
PROGRAM hello

USE mpi

INTEGER err, rank, size


    CALL MPI_Init(err)
   
    CALL MPI_Comm_rank (MPI_COMM_WORLD,rank, ierr)
   
    CALL MPI_Comm_size (MPI_COMM_WORLD,size, ierr)

    WRITE(*,*) 'Hello from processor ', rank, ' of ', size
    
    CALL  MPI_Finalize(err)

END PROGRAM hello
